src/main/scala/org/broadinstitute/dsde/firecloud/core/AgoraPermissionHandler.scala

```diff
@@ -96,26 +93,19 @@ class AgoraPermissionActor (requestContext: RequestContext) extends Actor with F
             try {
               val agoraPermissions = unmarshal[List[AgoraPermission]].apply(response)
               val fireCloudPermissions = agoraPermissions.map(x => x.toFireCloudPermission)
-              context.parent ! RequestComplete(OK, fireCloudPermissions)
-              context stop self
+              parent ! RequestComplete(OK, fireCloudPermissions)
             } catch {
               // TODO: more specific and graceful error-handling
               case e: Exception =>
-                context.parent ! RequestCompleteWithErrorReport(InternalServerError, "Failed to interpret methods " +
+                parent ! RequestCompleteWithErrorReport(InternalServerError, "Failed to interpret methods " +
                   "server response: " + e.getMessage)
-                context stop self
             }
           case x =>
-            context.parent ! RequestCompleteWithErrorReport(x, response.entity.asString)
-            context stop self
+            parent ! RequestCompleteWithErrorReport(x, response.entity.asString)
         }
       case Failure(e) =>
-        context.parent ! RequestCompleteWithErrorReport(StatusCodes.InternalServerError, e.getMessage)
-        context stop self
+        parent ! RequestCompleteWithErrorReport(StatusCodes.InternalServerError, e.getMessage)
```

Comment:
My recommendation would be to use the pipeTo pattern:
```
import akka.pattern.pipe
future pipeTo actor
```

instead of passing in the actor to pass the message to, handleAgoraResponse should return Future[PerRequestMessage] then your code looks like
```
handleAgoraResponse(context.parent, permissionListFuture) pipeTo context.parent
```
this avoid the whole "close over..." business entirely AND forces you to return something useful from handleAgoraResponse